edge/pkg/clusterd/mission_deployer.go

```diff
@@ -0,0 +1,386 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+package clusterd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	yaml "gopkg.in/yaml.v2"
+	"os"
+	"path/filepath"
+	"strings"
+
+	edgeclustersv1 "github.com/kubeedge/kubeedge/cloud/pkg/apis/edgeclusters/v1"
+	crdClientset "github.com/kubeedge/kubeedge/cloud/pkg/client/clientset/versioned"
+	"github.com/kubeedge/kubeedge/pkg/apis/componentconfig/edgecore/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	COMMAND_TIMEOUT_SEC  = 10
+	MISSION_CRD_FILE     = "mission_v1.yaml"
+	EDGECLUSTER_CRD_FILE = "edgecluster_v1.yaml"
+	LOCAL_EDGE_CLUSTER   = "local_edge_cluster"
+	STATUS_NO_MATCH      = "not match"
+)
+
+var DistroToKubectl = map[string]string{
+	"arktos": "kubectl/arktos/kubectl",
+	"k8s":    "kubectl/vanilla/kubectl",
+}
+
+type MissionDeployer struct {
+	ClusterName    string
+	ClusterLabels  map[string]string
+	KubeDistro     string
+	KubeconfigFile string
+	KubectlCli     string
+	MissionMatch   map[string]bool
+	CrdClient      *crdClientset.Clientset
+}
+
+//NewMissionDeployer creates new mission deployer object
+func NewMissionDeployer(clusterdConfig *v1alpha1.Clusterd) *MissionDeployer {
+
+	// No need to check the clusterdConfig, as it was checked during the clusterd initialization
+	basedir, _ := filepath.Abs(filepath.Dir(os.Args[0]))
+
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", clusterdConfig.Kubeconfig)
+	if err != nil {
+		klog.Fatalf("Failed to build config, err: %v", err)
+	}
+
+	crdKubeConfig := rest.CopyConfig(kubeConfig)
+	crdKubeConfig.ContentType = runtime.ContentTypeJSON
+
+	return &MissionDeployer{
+		ClusterName:    clusterdConfig.Name,
+		ClusterLabels:  clusterdConfig.Labels,
+		KubeDistro:     clusterdConfig.KubeDistro,
+		KubeconfigFile: clusterdConfig.Kubeconfig,
+		KubectlCli:     filepath.Join(basedir, DistroToKubectl[clusterdConfig.KubeDistro]),
+		MissionMatch:   map[string]bool{},
+		CrdClient:      crdClientset.NewForConfigOrDie(crdKubeConfig),
+	}
+}
+
+func (m *MissionDeployer) ApplyMission(mission *edgeclustersv1.Mission) error {
```

Comment:
word "Apply" here can be a bit confusing. Maybe use "Update" instead

Reply:
Actually, I think "Apply" is a better term. The action here could be "Create" or "Update".